### PR TITLE
Backend: Add blossom and make source checker show version again

### PR DESCRIPTION
**Pull request by ThatGravyBoat**
## What
Adds blossom to do token replacement in code. 
Adds back the version number to the security error screen.
Changes the hardcoded version number in SkyhanniMod to the token (should reduce amount of merge conflicts)

## Changelog Technical Details
+ Added Blossom Gradle plugin for token replacement in code. - ThatGravyBoat
    * `@MOD_VERSION@` is now a token that will be replaced.

### CODE REVIEW

**Review by CalMWolfs — APPROVED** [review]
Worked for me

File: settings.gradle.kts [hunk 1]
```diff
@@ -14,6 +14,7 @@ pluginManagement {
         maven("https://repo.nea.moe/releases")
         maven("https://repo.sk1er.club/repository/maven-releases/")
         maven("https://maven.deftu.dev/releases")
+        maven("https://repo.polyfrost.org/releases")
```
Comment (hannibal002) [inline settings.gradle.kts]: is there no other repo to get this from?

**Review by hannibal002 — CHANGES_REQUESTED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (ThatGravyBoat) [reply settings.gradle.kts]: wdym? this is the repo in which has a fork that supports the newer kotlin version as they changed how it works in 2.x but the fix wasnt backported to 1.x

Comment (ThatGravyBoat) [reply settings.gradle.kts]: This is the changes that Polyfrost made https://github.com/KyoriPowered/blossom/compare/master...Polyfrost:blossom:master

Comment (ThatGravyBoat) [reply settings.gradle.kts]: if the issue is about uptime of polyfrost infrastructure, I can put a mirror of it on my own maven (which is already used for the RPC)

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**